File: libs/power-log-parser/src/lib/handlers/hide-entity-handler.ts
import type { Helper } from '../helper';
import { Action, Game, HideEntity, Node } from '../models';
import { Regexes } from '../regexes';
import type { ParserState } from '../state/parser-state';

export class HideEntityHandler {
	static Handle(
		timestamp: string,
		data: string,
		state: ParserState,
		indentLevel: number,
		helper: Helper,
	): boolean {
		const match = Regexes.ActionHideEntityRegex.exec(data);
		if (match) {
			const rawEntity = match[1];
			const tagName = match[2];
			const value = match[3];
			const entity = helper.ParseEntity(rawEntity);
			const zone = helper.ParseTag(tagName, value);

			const hideEntity = new HideEntity();
			hideEntity.Entity = entity;
			hideEntity.Zone = zone.Value;
			hideEntity.TimeStamp = timestamp;

			state.UpdateCurrentNode(Game, Action);

			if (state.Node!.Type === Game) {
				(state.Node!.Object as Game).AddData(hideEntity);
			} else if (state.Node!.Type === Action) {
				(state.Node!.Object as Action).Data.push(hideEntity);
			} else {
				throw new Error('Invalid node: ' + state.Node!.Type);
			}

			const newNode = new Node(HideEntity, hideEntity, indentLevel, state.Node, data);
			state.CreateNewNode(newNode);
			state.Node = newNode;
			return true;
		}
		return false;
	}
}

File: tools/linux-helpers/hs-converter/HearthstoneReplays/Parser/Handlers/DataHandler/HideEntityHandler.cs
using HearthstoneReplays.Enums;
using HearthstoneReplays.Parser.ReplayData.Entities;
using HearthstoneReplays.Parser.ReplayData.GameActions;
using HearthstoneReplays.Parser.ReplayData;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Action = HearthstoneReplays.Parser.ReplayData.GameActions.Action;

namespace HearthstoneReplays.Parser.Handlers
{
    internal class HideEntityHandler
    {
        public static bool HandleHideEntity(DateTime timestamp, string data, ParserState state, int indentLevel, Helper helper)
        {
            var match = Regexes.ActionHideEntityRegex.Match(data);
            if (match.Success)
            {
                var rawEntity = match.Groups[1].Value;
                var tagName = match.Groups[2].Value;
                var value = match.Groups[3].Value;
                var entity = helper.ParseEntity(rawEntity);
                var zone = helper.ParseTag(tagName, value);

                var hideEntity = new HideEntity { Entity = entity, Zone = zone.Value, TimeStamp = timestamp };
                state.UpdateCurrentNode(typeof(Game), typeof(Action));

                if (state.Node.Type == typeof(Game))
                    ((Game)state.Node.Object).AddData(hideEntity);
                else if (state.Node.Type == typeof(Action))
                    ((Action)state.Node.Object).Data.Add(hideEntity);
                else
                    throw new Exception("Invalid node: " + state.Node.Type);

                var newNode = new Node(typeof(HideEntity), hideEntity, indentLevel, state.Node, data);
                state.CreateNewNode(newNode);
                state.Node = newNode;
                return true;
            }
            return false;
        }
    }
}
